### PR TITLE
Adding solidity workflow.

### PR DESCRIPTION
**Pull request by ebadiere**
Adding new solidity tests GH test workflow.  Related issues: #430 

Fixes #

### CODE REVIEW

File: .github/workflows/solidity-tests.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Solidity Tests CI
+
+on:
```
Comment (Nana-EC) [inline .github/workflows/solidity-tests.yml:3]: Make this manual. There will be a lot of tests so we don't want to run them always so it should be from a workflow kick off

File: .github/workflows/solidity-tests.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  group: tests-${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  contents: write
+  checks: write
+  pull-requests: write
+
+jobs:
+  BlockInfo:
```
Comment (Nana-EC) [inline .github/workflows/solidity-tests.yml:20]:
```suggestion
  SolidityEquivalence:
```

File: .github/workflows/solidity-tests.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    name: BlockInfo Test Suite
+    uses: ./.github/workflows/test-workflow.yml
+    with:
+      testfilter: BlockInfo
```
Comment (Nana-EC) [inline .github/workflows/solidity-tests.yml:24]: Let's use a reusable tag
```suggestion
      testfilter: @solidityequiv
```

File: .github/workflows/solidity-tests.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+name: Solidity Tests CI
```
Comment (Nana-EC) [inline .github/workflows/solidity-tests.yml:1]: nit:
```suggestion
name: Solidity Equivalence Tests CI
```

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Nice.
Some suggestions to make it more general

File: .github/workflows/solidity-tests.yml (hunk 1 above)
Comment (ebadiere) [reply .github/workflows/solidity-tests.yml:3]: Done.

File: .github/workflows/solidity-tests.yml (hunk 2 above)
Comment (ebadiere) [reply .github/workflows/solidity-tests.yml:16]: By doing this I assume we are going to run all of the solidity equivalence tests in one job.  Is that what we want?  I am not sure that's scalable.

File: .github/workflows/solidity-tests.yml (hunk 3 above)
Comment (ebadiere) [reply .github/workflows/solidity-tests.yml:20]: Done, but I still question putting all tests in one umbrella job.

File: .github/workflows/solidity-tests.yml (hunk 4 above)
Comment (ebadiere) [reply .github/workflows/solidity-tests.yml:1]: Done.

File: .github/workflows/solidity-tests.yml (hunk 2 above)
Comment (Nana-EC) [reply .github/workflows/solidity-tests.yml:16]: You can break them out but it shouldn't be by file.
We can group them in larger buckets.
What would you suggest?

Comment (ebadiere) [reply .github/workflows/solidity-tests.yml:16]: I guess the term "SolidityEquivalence" is what I am questioning here.  I think it makes sense to start with this, so I'll make the update, but the term seems to be an umbrella term for this whole effort which will result in many more tests.

Comment (ebadiere) [reply .github/workflows/solidity-tests.yml:16]: Done.

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**